core/components/atoms/input/Input.tsx

```diff
@@ -192,6 +192,8 @@ export const Input = React.forwardRef<HTMLInputElement, InputProps>((props, forw
       [`Input--${size}`]: size,
       ['Input--disabled']: disabled || readOnly,
       ['Input--error']: error,
+      ['Input--leftIcon']: size !== 'tiny' && icon,
+      ['Input--rightIcon']: !disabled && (info || ((actionIcon || onClear) && (value || defaultValue))),
```

Comment:
Please follow the BEM convention for className. Instead of using `Input--leftIcon`  please use `Input-withIcon--left`